### ISSUE TITLE
StorageCipher initialization failed

### ISSUE THREAD

**Issue by Droidlover** [mogol/flutter_secure_storage#159]
Hi There,

First thanks for latest 3.3.4 update. Nice to see this is still under development and active.

Now issue I am facing is that on Android 9,I get following error : 

StorageCipher initialization failed
java.security.ProviderException: Failed to load generated key pair from keystore

Have tested on Androi 6,7,10. Issue occurs only on andoir 9/Pie.

I am simply using  :  
```
final storage = new FlutterSecureStorage();
await storage.read(key: 'sortedEncryptionKey');
```

Following are the logs  :  

```
D/FlutterSecureStoragePl(31078): Initializing StorageCipher
I/fluttersecurestorage(31078): Creating keys!
I/fluttersecurestorage(31078): Initializing
I/fluttersecurestorage(31078): Generating key pair
E/FlutterSecureStoragePl(31078): StorageCipher initialization failed
E/FlutterSecureStoragePl(31078): java.security.ProviderException: Failed to load generated key pair from keystore
E/FlutterSecureStoragePl(31078): 	at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.loadKeystoreKeyPair(AndroidKeyStoreKeyPairGeneratorSpi.java:530)
E/FlutterSecureStoragePl(31078): 	at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.generateKeyPair(AndroidKeyStoreKeyPairGeneratorSpi.java:472)
E/FlutterSecureStoragePl(31078): 	at java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:727)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.ciphers.RSACipher18Implementation.createKeys(RSACipher18Implementation.java:189)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.ciphers.RSACipher18Implementation.createRSAKeysIfNeeded(RSACipher18Implementation.java:120)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.ciphers.RSACipher18Implementation.<init>(RSACipher18Implementation.java:38)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.ciphers.StorageCipher18Implementation.<init>(StorageCipher18Implementation.java:31)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin.ensureInitStorageCipher(FlutterSecureStoragePlugin.java:75)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin.access$000(FlutterSecureStoragePlugin.java:29)
E/FlutterSecureStoragePl(31078): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin$MethodRunner.run(FlutterSecureStoragePlugin.java:183)
E/FlutterSecureStoragePl(31078): 	at java.lang.Thread.run(Thread.java:764)
E/FlutterSecureStoragePl(31078): Caused by: java.security.UnrecoverableKeyException: Failed to obtain information about key
E/FlutterSecureStoragePl(31078): 	at android.security.keystore.AndroidKeyStoreProvider.getKeyCharacteristics(AndroidKeyStoreProvider.java:234)
E/FlutterSecureStoragePl(31078): 	at android.security.keystore.AndroidKeyStoreProvider.loadAndroidKeyStoreKeyPairFromKeystore(AndroidKeyStoreProvider.java:300)
E/FlutterSecureStoragePl(31078): 	at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.loadKeystoreKeyPair(AndroidKeyStoreKeyPairGeneratorSpi.java:521)
E/FlutterSecureStoragePl(31078): 	... 10 more
E/FlutterSecureStoragePl(31078): Caused by: android.security.KeyStoreException: -22
E/FlutterSecureStoragePl(31078): 	at android.security.KeyStore.getKeyStoreException(KeyStore.java:839)
E/FlutterSecureStoragePl(31078): 	at android.security.keystore.AndroidKeyStoreProvider.getKeyCharacteristics(AndroidKeyStoreProvider.java:236)
E/FlutterSecureStoragePl(31078): 	... 12 more
```

I will be very thankful if i get some help here.

**Comment by koskimas** [comment]
@Droidlover Please try the updated version 3.3.5. It fixed a bunch of android issues.

**Comment by haydgately** [comment]
Still broken on 3.3.2 @koskimas for me I'm afraid!

**Comment by bananenwilly** [comment]
`D/FlutterSecureStoragePl(11858): Read: key not present => Skip ensureInitStorageCipher and return null
D/FlutterSecureStoragePl(11858): Initializing StorageCipher
I/fluttersecurestorage(11858): Creating keys!
I/fluttersecurestorage(11858): Initializing
I/fluttersecurestorage(11858): Generating key pair
E/KeyStore(11858): generateKeyInternal failed on request -68
E/FlutterSecureStoragePl(11858): StorageCipher initialization failed
E/FlutterSecureStoragePl(11858): java.security.ProviderException: android.security.keystore.StrongBoxUnavailableException: Failed to generate key pair
E/FlutterSecureStoragePl(11858): 	at android.security.keystore.AndroidKeyStoreKeyPairGeneratorSpi.generateKeyPair(AndroidKeyStoreKeyPairGeneratorSpi.java:487)
E/FlutterSecureStoragePl(11858): 	at java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:727)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.ciphers.RSACipher18Implementation.createKeys(RSACipher18Implementation.java:202)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.ciphers.RSACipher18Implementation.createRSAKeysIfNeeded(RSACipher18Implementation.java:121)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.ciphers.RSACipher18Implementation.<init>(RSACipher18Implementation.java:39)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.ciphers.StorageCipher18Implementation.<init>(StorageCipher18Implementation.java:31)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin.ensureInitStorageCipher(FlutterSecureStoragePlugin.java:74)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin.access$000(FlutterSecureStoragePlugin.java:30)
E/FlutterSecureStoragePl(11858): 	at com.it_nomads.fluttersecurestorage.FlutterSecureStoragePlugin$MethodRunner.run(FlutterSecureStoragePlugin.java:188)
E/FlutterSecureStoragePl(11858): 	at android.os.Handler.handleCallback(Handler.java:938)
E/FlutterSecureStoragePl(11858): 	at android.os.Handler.dispatchMessage(Handler.java:99)
`
same problem on Android S preview arm abi  

version 4.1.0

**Comment by mnesarco** [comment]
I have this same problem with 4.1.0

**Comment by juliansteenbakker** [comment]
Can someone check if this issue still persists with version 4.2.0?

**Comment by bananenwilly** [comment]
4.2.0 looks promising so far

**Comment by PascalMarechal** [comment]
@juliansteenbakker, where do you write this piece of code?

**Comment by juliansteenbakker** [comment]
@PascalMarechal See the example app: https://github.com/mogol/flutter_secure_storage/blob/5f4880d9ed05de64274ce8b88572d5706a091c20/flutter_secure_storage/example/lib/main.dart#L70

**Comment by duzenko** [comment]
> 
> 
> @PascalMarechal See the example app:
> 
> https://github.com/mogol/flutter_secure_storage/blob/5f4880d9ed05de64274ce8b88572d5706a091c20/flutter_secure_storage/example/lib/main.dart#L70

The named parameter 'encryptedSharedPreferences' isn't defined.
Try correcting the name to an existing named parameter's name, or defining a named parameter with the name 'encryptedSharedPreferences'.

**Comment by juliansteenbakker** [comment]
Which version of flutter_secure_storage are you using? @duzenko

**Comment by duzenko** [comment]
latest 4.2.1

**Comment by juliansteenbakker** [comment]
All new functions are available in the beta release v5.0.0-beta.5
